### PR TITLE
update: Update apps from both system and user installations

### PR DESCRIPTION
**Pull request by alexlarsson**
Unless you manually specify an installation it looks for updates in all the standard installations.

This fixes https://github.com/flatpak/flatpak/issues/282

### CODE REVIEW

File: app/flatpak-transaction.c [hunk 1]
```diff
@@ -157,6 +157,12 @@ flatpak_transaction_operation_free (FlatpakTransactionOp *self)
   g_free (self);
 }
 
+gboolean
+flatpak_transaction_is_emptry (FlatpakTransaction  *self)
```
Comment (mwleeds) [inline app/flatpak-transaction.c]: Is this a typo?

File: doc/flatpak-update.xml [hunk 2]
```diff
@@ -76,8 +76,9 @@
             version, with the --commit option.
         </para>
         <para>
-            Unless overridden with the --user or the --installation option, this command updates
-            the default system-wide installation.
+            Unless overridden with the --user, --system or -installation option, this command updates
+            any matching refs in the standard the default system-wide installation and the
```
Comment (mwleeds) [inline doc/flatpak-update.xml]: Take out "the default"?

File: doc/flatpak-update.xml [hunk 3]
```diff
@@ -76,8 +76,9 @@
             version, with the --commit option.
         </para>
         <para>
-            Unless overridden with the --user or the --installation option, this command updates
-            the default system-wide installation.
+            Unless overridden with the --user, --system or -installation option, this command updates
```
Comment (mwleeds) [inline doc/flatpak-update.xml]: missing hyphen before `-installation`

File: app/flatpak-builtins-update.c [hunk 4]
```diff
@@ -67,51 +67,72 @@ static GOptionEntry options[] = {
...
-      if (!res)
-        return FALSE;
-      flatpak_terminal_progress_end (&terminal_progress);
+      gboolean found = FALSE;
+
+      for (j = 0; j < dirs->len; j++)
+        {
+          FlatpakDir *dir = g_ptr_array_index (dirs, j);
+
+          if (flatpak_dir_has_remote (dir, remote))
```
Comment (mwleeds) [inline app/flatpak-builtins-update.c]: Wouldn't it make sense to use `flatpak_resolve_duplicate_remotes` like the other commands do?

Comment (alexlarsson) [reply app/flatpak-builtins-update.c]: The new behaviour in general in this commit is that if you specify no --user or --system, then all matching things are updated. For instance, flatpak update org.gnome.gedit will update both the system and the user gedit (if installed), and we error only if the specifice ref doesn't exist in any of the dirs. Similarly, flatpak update --appstream updates appstream data for all remotes in both system and user.

So, the question is, should flatpak update --appstream flathub update appstream in both user and system to match the above behaviour, or should it stop at the user appstream?

I chose to match the other update behaviour, given that an update (especially of appdata) is a pretty "safe" operation, whereas accidentally not updating when you think it did can be a pain.

I guess the overall question is whether the "update both system and user" is the right behaviour for update though. Opinions?

**Comment by alexlarsson** [comment]
Should be fixed now

**Comment by alexlarsson** [comment]
I'll merge this for now, because i want to do something else that depends on it.

**Comment by alexlarsson** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**